mmedit/models/backbones/sr_backbones/liif_net.py

```diff
@@ -0,0 +1,259 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.runner import load_checkpoint
+
+from mmedit.datasets.pipelines.utils import make_coord
+from mmedit.models.builder import build_backbone, build_component
+from mmedit.models.registry import BACKBONES
+from mmedit.utils import get_root_logger
+
+
+class LIIFNet(nn.Module):
+    """LIIF net for single image super-resolution.
+
+    Paper: Learning Continuous Image Representation with
+           Local Implicit Image Function
+
+    Args:
+        encoder (dict): Config for the generator.
+        imnet (dict): Config for the imnet.
+        local_ensemble (bool): Whether to use local ensemble. Default: True.
+        feat_unfold (bool): Whether to use feat unfold. Default: True.
+        cell_decode (bool): Whether to use cell decode. Default: True.
+        eval_bsize (int): Size of batched predict. Default: None.
+    """
+
+    def __init__(self,
+                 encoder,
+                 imnet,
+                 local_ensemble=True,
+                 feat_unfold=True,
+                 cell_decode=True,
+                 eval_bsize=None):
+        super().__init__()
+
+        self.local_ensemble = local_ensemble
+        self.feat_unfold = feat_unfold
+        self.cell_decode = cell_decode
+        self.eval_bsize = eval_bsize
+
+        # model
+        self.encoder = build_backbone(encoder)
+        imnet_in_dim = self.encoder.mid_channels
```

Comment:
There may be a problem if `self.encoder` has no `mid_channels`.

Reply:
self.encoder can be EDSR or RDN, all of them have `mid_channels`.

Reply:
Then we should add a value error? Now there is a chance that users use some other networks that contain no `mid_channels`.

Reply:
Let's keep status quo and add helpful message when issues flood in.